htdocs/class/xoopsmailer.php

```diff
@@ -560,7 +560,8 @@ public function setToGroups($group)
             if (strtolower(get_class($group)) === 'xoopsgroup') {
                 /* @var XoopsMemberHandler $member_handler */
                 $member_handler = xoops_getHandler('member');
-                $this->setToUsers($member_handler->getUsersByGroup($group->getVar('groupid'), true));
+				$users = $member_handler->getUsersByGroup($group->getVar('groupid'), true);
+                $this->setToUsers($users);										  
```

Comment:
First, this is a great catch. Thanks!

Looking at this, there is no good reason for XoopsMailer::setToUsers() to expect a pass by reference argument. Fixing that solves the issue this and several more like it. See #677